network/address_test.go

```diff
@@ -317,14 +317,22 @@ var selectPublicTests = []selectTest{{
 	},
 	-1,
 }, {
-	"a public name is preferred to an unknown or cloud local address",
+	"a public name is preferred to an unknown or cloud local address (no IPv4)",
+	[]network.Address{
+		network.NewScopedAddress("127.0.0.1", network.ScopeUnknown),
+		network.NewScopedAddress("fc00::1", network.ScopeCloudLocal),
+		network.NewScopedAddress("public.invalid.testing", network.ScopePublic),
+	},
+	2,
+}, {
+	"a cloud local IPv4 is preferred to a name,  unknown scope or cloud local address",
```

Comment:
If we have 2 addresses to pick from: both with ScopeCloudLocal, one IPv4Address, the other HostName, which one will be picked with the changes? I'd expect the latter, as hostnames should still be preferred over same-scoped IPv4 addresses, please double check and add a test for it, if there isnt' one already.